packages/typespec-vscode/src/emit/emit.ts

```diff
@@ -0,0 +1,335 @@
+import path from "path";
+import vscode, { Uri } from "vscode";
+import { Executable } from "vscode-languageclient/node.js";
+import { StartFileName } from "../const.js";
+import logger from "../log/logger.js";
+import { InstallationAction, npmDependencyType, NpmUtil } from "../npm-utils.js";
+import { getDirectoryPath } from "../path-utils.js";
+import { resolveTypeSpecCli } from "../tsp-executable-resolver.js";
+import {
+  getEntrypointTspFile,
+  onStdioError,
+  onStdioOut,
+  TraverseMainTspFileInWorkspace,
+} from "../typespec-utils.js";
+import { ExecOutput, isFile, spawnExecution } from "../utils.js";
+import { EmitQuickPickItem } from "./emit-quick-pick-item.js";
+import {
+  Emitter,
+  EmitterKind,
+  getRegisterEmitters,
+  getRegisterEmitterTypes,
+  PreDefinedEmitterPickItems,
+} from "./emitter.js";
+
+export async function doEmit(
+  context: vscode.ExtensionContext,
+  mainTspFile: string,
+  kind: EmitterKind,
+) {
+  if (!mainTspFile || !(await isFile(mainTspFile))) {
+    logger.info(
+      "Invalid typespec project. There is no main tsp file in the project. Generating Cancelled.",
+      [],
+      { showOutput: false, showPopup: true },
+    );
+    return;
+  }
+
+  const baseDir = getDirectoryPath(mainTspFile);
+
+  const toQuickPickItem = (e: Emitter): EmitQuickPickItem => {
+    return {
+      language: e.language,
+      package: e.package,
+      emitterKind: e.kind,
+      label: e.language,
+      detail: `Generate ${e.language} ${e.kind} code from ${e.package}`,
+      picked: false,
+      fromConfig: false,
+      iconPath: Uri.file(context.asAbsolutePath(`./icons/${e.language.toLowerCase()}.svg`)),
+    };
+  };
+
+  const registerEmitters = getRegisterEmitters(kind);
+  const all = [...registerEmitters].map((e) => toQuickPickItem(e));
+
+  const selectedEmitter = await vscode.window.showQuickPick<EmitQuickPickItem>(all, {
+    title: "Select a Language",
+    canPickMany: false,
+    placeHolder: "Pick a Language",
+    ignoreFocusOut: true,
+  });
+
+  if (!selectedEmitter) {
+    logger.info("No emitter selected. Generating Cancelled.", [], {
+      showOutput: false,
+      showPopup: false,
```

Comment:
false is the default value, you can just logger.info("...") to make your code shorter.